test/python/pulse/test_instructions.py

```diff
@@ -285,23 +606,90 @@ def setUp(self):
         self.duration = 4
         self.pulse_op = library.Waveform([1.0] * self.duration, name="test")
 
-    def test_play(self):
+    def test_play_legacy(self):
         """Test basic play instruction."""
-        play = instructions.Play(self.pulse_op, channels.DriveChannel(1))
+        play = instructions.Play(self.pulse_op, channel=channels.DriveChannel(1))
 
         self.assertIsInstance(play.id, int)
         self.assertEqual(play.name, self.pulse_op.name)
         self.assertEqual(play.duration, self.duration)
+        self.assertEqual(play.channel, channels.DriveChannel(1))
+        self.assertEqual(play.inst_target, channels.DriveChannel(1))
         self.assertEqual(
             repr(play),
             "Play(Waveform(array([1.+0.j, 1.+0.j, 1.+0.j, 1.+0.j]), name='test'),"
             " DriveChannel(1), name='test')",
         )
 
+    def test_play_new_model_target_frame(self):
+        """Test basic play instruction."""
+        play = instructions.Play(self.pulse_op, target=Qubit(0), frame=QubitFrame(0))
+
+        self.assertIsInstance(play.id, int)
+        self.assertEqual(play.name, self.pulse_op.name)
+        self.assertEqual(play.duration, self.duration)
+        self.assertEqual(play.inst_target, MixedFrame(Qubit(0), QubitFrame(0)))
+        self.assertEqual(play.channel, None)
+
+    def test_play_new_model_mixed_frame(self):
+        """Test basic play instruction."""
+        mf = MixedFrame(Qubit(0), QubitFrame(0))
+        play = instructions.Play(self.pulse_op, mixed_frame=mf)
+
+        self.assertIsInstance(play.id, int)
+        self.assertEqual(play.name, self.pulse_op.name)
+        self.assertEqual(play.duration, self.duration)
+        self.assertEqual(play.inst_target, mf)
+        self.assertEqual(play.channel, None)
+
     def test_play_non_pulse_ch_raises(self):
         """Test that play instruction on non-pulse channel raises a pulse error."""
         with self.assertRaises(exceptions.PulseError):
-            instructions.Play(self.pulse_op, channels.AcquireChannel(0))
+            instructions.Play(self.pulse_op, channel=channels.AcquireChannel(0))
+
+    def test_play_arguments_validation(self):
```

Comment:
I see this pattern everywhere. I personally prefer having a single validation per test. What do you think of using following pattern?

```python
from ddt import ddt, named_data, unpack

@ddt
class  ...
    @named_data(
        ["only_frame", None, None, QubitFrame(0), None],
        ["only_target", None, Qubit(0), None, None],
        ...
    )
    @unpack
    def test_invalid_construction(self, channel, target, frame, mixed_frame):
        with self.assertRaises(exceptions.PulseError):
            instructions.Play(
                self.pulse_op, 
                channel=channel, 
                target=target, 
                frame=frame, 
                mixed_frame=mixed_frame,
            )
    
```

This pattern is almost equivalent to define a dedicated test function for each condition. `named_data` decorator is really developer friendly since it tells you the test condition in human readable manner.

Reply:
Nifty trick!